### ISSUE TITLE
[sub-pause] Make it work for .idx/.sub subtitles

### ISSUE THREAD

**Issue by Des-Nerger** [Ben-Kerman/mpv-sub-scripts#8]
Since .idx/.sub subtitles don't have the "sub-text" property, the current implementation doesn't work for them.
How about observing "sub-start" or "sub-end" instead of "sub-text"?
I tried a quick hack:
```diff
-		mp.observe_property("sub-text", "string", handle_sub_text_change)
+		mp.observe_property("sub-start", "string", handle_sub_text_change)
```
, and it did the job for me.

**Comment by Ben-Kerman** [comment]
That's something I already wanted to implement. It only became possible with the latest version of mpv so I haven't gotten around to it yet, but I'll change the script to use sub-end instead of sub-text at some point

**Comment by Thermospore** [comment]
Would this make it work for .sup files too?

eg: https://kitsunekko.net/subtitles/japanese/K-ON!/[philosophy-raws][K-ON!!][01][BDRIP][Hi10P%20FLAC][1920X1080].SUP%201-9.zip

**Comment by Des-Nerger** [comment]
Yes, I just tried `.sup`s for some other anime series I had and they worked.
Although when I also tried `sub-pause.lua` on a `.sup` for Akira: https://kitsunekko.net/subtitles/japanese/Akira/[RUELL-Raws]%20AKIRA%20(BD%201280x688%20x264%20DualAAC(JP+EN)%205.1ch%20JP+ENSub)%20[E1F89E0A].rar, even though it auto-pauses at starts of lines flawlessy, auto-pauses at ends of lines often occur too early. Maybe @Ben-Kerman knows why. Perhaps related to how the script predicts times for those ends.

**Comment by Ben-Kerman** [comment]
Should be fixed with 065b856 and 2df39cf for both scripts, respectively. Sorry it took so long, @Des-Nerger @Thermospore

As for the issue with the Akira PGS subs, that seems to be because for some reason mpv reports several different values of `sub-start` and `sub-end` for each line. I suspect it's because those lines are actually split like that in the PGS stream for some reason. It works with other SUP files I tried. If this is a frequent problem with other PGS subs as well feel free to open a dedicated issue
